plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptions/Utils.php

```diff
@@ -154,4 +154,25 @@ public static function render_block_with_context( $block, $context ) {
 		// Render with dynamic set to false to prevent calling render_callback.
 		return $new_block->render( array( 'dynamic' => false ) );
 	}
+
+	/**
+	 * Generates a consistent ID for a product's quantity input.
+	 *
+	 * @param \WC_Product $product The product object.
+	 * @return string The generated ID.
+	 */
+	public static function get_quantity_input_id( $product ) {
+		return 'quantity_' . md5( $product->get_id() . '_' . $product->get_name() );
+	}
+
+	/**
+	 * Removes the label from quantity input HTML.
+	 *
+	 * @param string $quantity_html The quantity input HTML.
+	 * @return string The quantity input HTML without the label.
+	 */
+	public static function remove_quantity_label( $quantity_html ) {
+		// Remove the label from the quantity input.
+		return preg_replace( '/<label[^>]*>.*?<\/label>/s', '', $quantity_html );
+	}
```

Comment:
I think we don't need to extract this function now. We are only using it in one place, so this can be embedded there. If we see the need to reuse this logic, we can extract it later.

Reply:
Done in 9e67a09023b0a20ef15c18e42b425c3b8ba0f8a0.